src/vs/workbench/contrib/notebook/browser/view/cellParts/cellContextKeys.ts

```diff
@@ -101,7 +103,7 @@ export class CellContextKeyManager extends Disposable {
 
 		if (element instanceof CodeCellViewModel) {
 			this.elementDisposables.add(element.onDidChangeOutputs(() => this.updateForOutputs()));
-			this.elementDisposables.add(element.cellDiagnostics.onDidDiagnosticsChange(() => this.updateForDiagnostics()));
+			this.elementDisposables.add(this._notebookCellDiagnosticsService.onDidDiagnosticsChange(() => this.updateForDiagnostics()));
```

Comment:
I'm wondering if we can introduce the concept of `NotebookEditorWidget.getCellContextKeyManager()`. Once we have a cell context key manager, it can expose `ContextKeyManager.createContextKey(): IDisposable`